File: SMI-SDK/Java SDK/src/pt/sapo/gis/trace/ITrace.java
package pt.sapo.gis.trace;

import java.util.Map;
import java.util.Set;
import pt.sapo.gis.trace.configuration.SeverityType;

/**
* @author GIS Team 
 * Trace interface
 */
public interface ITrace {    
    void trace(Entry entry);
    Entry beginEntry();
    Entry beginEntry(String description, SeverityType severity);
    Entry beginEntry(String description, SeverityType severity, Map<String, Object> properties);
    Entry beginEntry(Entry entry);
    void endEntry();
    void finish();
    Set<SeverityType> getSeverityTypes();
    
}

File: SMI-SDK/.NET SDK/Library/gis/trace/ITrace.cs
using System;
using pt.sapo.gis.trace.configuration;
using System.Collections.Generic;
namespace pt.sapo.gis.trace
{

    /**
    * @author GIS Team 
     * Trace interface
     */
    public interface ITrace
    {
        void TraceEntry(Entry entry);
        Entry BeginEntry();
        Entry BeginEntry(String description, severityType severity);
        Entry BeginEntry(String description, severityType severity, IDictionary<String, Object> properties);
        Entry BeginEntry(Entry entry);
        void EndEntry();
        void Finish();
        ISet<severityType> GetSeverityTypes();
    }
}
